packages/core/src/awsService/cloudWatchLogs/commands/tailLogGroup.ts

```diff
@@ -99,6 +103,24 @@ function formatLogEvent(logEvent: LiveTailSessionLogEvent): string {
     return line
 }
 
+//Auto scroll visible LiveTail session editors if the end-of-file is in view.
+//This allows for newly added log events to stay in view.
+function getTextEditorsToScroll(document: vscode.TextDocument): vscode.TextEditor[] {
+    return vscode.window.visibleTextEditors.filter((editor) => {
+        const isEditorForSession = editor.document === document
```

Comment:
I think this is only used once, should you just inline this on line 111

Reply:
I could do that!

The reason I wrote this as a function was not so much about reusability, but more about readability. The code going into `handleSessionStream` is growing in size and complexity. I worry that as we do more per stream update that it will become less clear what "actions" we take. Having this defined as a function makes it easier to see that per stream update that we need to 'getTextEditorsToScroll`, then update the document, then scroll it. While abstracting away those implementations.

Reply:
Oh yeah, I thought that make sense. I was more just commenting on doing:
```
if (editor.document !== document) {
...
}
```
vs
```
const isEditorForSession = editor.document === document
if (!isEditorForSession) {

}
```

Reply:
Oh haha I misunderstood 😅 I'll change this